pkg/config/config.go

```diff
@@ -35,7 +35,7 @@ type ComponentSettings struct {
 	Ref *string `yaml:"Ref,omitempty"`
 
 	// Type is type of component source: git/local/binary
-	SourceType *occlient.CreateType `yaml:"SourceType,omitempty"`
+	SourceType *SrcType `yaml:"SourceType,omitempty"`
```

Comment:
@anmolbabu , can you please move struct `LocalConfig` to separate `types.go`, it becomes easy to manage structs

Reply:
like how it's done with URL and storage

Reply:
May be out of scope this PR? ;)
I'll move them in a follow up PR(The structs was added by a different PR anyway)
But I'll add it to my checklist in https://github.com/redhat-developer/odo/pull/1360#issuecomment-469179197 which we'll use to create a followup issue to track any new changes(but bugs and any code cleanups that are essential for this PR to be in will be addressed here as per my discussion with @kadel yesterday)

Reply:
follow up works, not necessary with this PR